### PR TITLE
REF: dispatch Series.rank to EA

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

Test coverage is severely lacking here.

Maybe could implement the general case in terms of argsort?  cc @TomAugspurger @jorisvandenbossche

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -8494,19 +8494,32 @@ def rank(
...
+            if isinstance(values, ExtensionArray):
+                ranks = values._rank(
+                    axis=axis,
+                    method=method,
+                    ascending=ascending,
+                    na_option=na_option,
+                    pct=pct,
+                )
+            else:
+                ranks = algos.rank(
```
Comment (jreback) [inline pandas/core/generic.py:8513]: can you not just do this indiretion *in* algos?

Comment (jbrockmendel) [reply pandas/core/generic.py:8513]: either way i guess

Comment (jbrockmendel) [reply pandas/core/generic.py:8513]: i take it back, doing it in algos causes a RecursionError

Comment (jreback) [reply pandas/core/generic.py:8513]: kk i see what you did for mode so maybe this ok

**Comment by jreback** [comment]
looks ok to me

**Pull request merged**